python/dolfinx/fem/forms.py

```diff
@@ -56,22 +59,22 @@ def code(self) -> typing.Union[str, None]:
 
     @property
     def rank(self) -> int:
-        return self._cpp_object.rank
+        return self._cpp_object.rank  # type: ignore
 
     @property
     def function_spaces(self) -> typing.List[FunctionSpace]:
         """Function spaces on which this form is defined"""
-        return self._cpp_object.function_spaces
+        return self._cpp_object.function_spaces  # type: ignore
 
     @property
     def dtype(self) -> np.dtype:
         """Scalar type of this form"""
-        return self._cpp_object.dtype
+        return self._cpp_object.dtype  # type: ignore
 
     @property
-    def mesh(self) -> Mesh:
+    def mesh(self) -> typing.Tuple[_cpp.fem.Mesh_float32, _cpp.fem.Mesh_float64]:
```

Comment:
Reminder to self: another `tuple` vs `union` doubt.

This is actually what triggered me to look into this PR. `multiphenicsx` CI started to fail, but I can't really see how the `mesh` property is supposed to be a tuple containing a float32 mesh and a `float64` one.